lib/validations.js

```diff
@@ -340,7 +340,20 @@ function validateCustom(attr, conf, err, options, done) {
     done = options;
     options = {};
   }
-  conf.customValidator.call(this, err, done);
+
+  if (conf.customValidator.length <= 1) {
```

Comment:
Actually: when defining a validator, user can provide `sync` flag to tell the framework whether the validator function is sync or async. Perhaps we can add another flag, e.g. `promise` to distinguish between "old" custom validators using callbacks and the new Promise-based flavour?

Just an idea to consider.